go/fn/run.go

```diff
@@ -27,7 +27,14 @@ import (
 // AsMain reads the resourceList in yaml format from stdin, evaluates the
 // function and write the updated resourceList in yaml to stdout. Errors if any
 // will be printed to stderr.
-func AsMain(p ResourceListProcessor) error {
+func AsMain(input interface{}) error {
+	var p ResourceListProcessor
+	switch input := input.(type) {
+	case FunctionRunner:
+		p = runnerProcessor{fnRunner: input}
+	case ResourceListProcessorFunc:
+		p = input
+	}
```

Comment:
You need to add a default case here.
If a user passes a wrong type, we should return an error.

Reply:
good catch! Thanks!